### ISSUE TITLE
Bundle custom widget lib for Qt Designer 

### ISSUE THREAD

**Issue by PeterPetrik** [qgis/QGIS-Mac-Packager#3]
https://github.com/lutraconsulting/qgis-mac-packager/issues/80

It would be nice to have the custom widgets available for Qt Designer

Run cmake with -DWITH_CUSTOMWIDGETS=ON
This will place the lib in QT_PLUGINS_DIR
Then the question is how any existing Qt Designer would find the lib?
a) The user would have to add the path to the lib before starting Qt Designer
b) or we ship Qt Designer in the bundle?

**Comment by M-Rick** [comment]
Yes it would we great to get it. I have plenty of custom forms not working anymore in QGIS3 that have to be updated with custom widgets. Currently the only way is by running QGIS inside a Linux VM. Thats's not really convenient.

**Comment by 3nids** [comment]
To me, the best approach seems to copy the Designer.app directly into QGIS bundle.

Following this [answer](https://stackoverflow.com/a/27360031/1548052), it seems possible

* [ ] Run cmake with -DWITH_CUSTOMWIDGETS=ON
* [ ] Check that the lib is in QT_PLUGINS_DIR
* [ ] Copy /usr/local/Cellar/qt/5.x.y/libexec/Designer.app into the bundle

@PeterPetrik could you as a quick try, just copy Designer.app in the bundle? 
I am not familiar with the code...

**Comment by M-Rick** [comment]
It is the way it's done with windows packages.

**Comment by M-Rick** [comment]
QtDesigner 4 works with only those items.
[QtDesigner 4.zip](https://github.com/qgis/QGIS-Mac-Packager/files/3962150/QtDesigner.4.zip)

**Comment by PeterPetrik** [comment]
should be part or tomorrow's nightly2 builds

**Comment by M-Rick** [comment]
I tried the latest nightly2 which is qgis_nightly2_master_20200723_032054.

I did find Qt Designer packaged inside at QGIS.app/Contents/MacOS/Designer.app

But when launching it, it doesn't include the QGIS custom widgets. But at least it works well.

![Capture d’écran 2020-07-23 à 15 30 36](https://user-images.githubusercontent.com/2520948/88292130-8516ac00-ccf9-11ea-8073-69d6a0a05bac.png)

**Comment by PeterPetrik** [comment]
maybe related https://github.com/qgis/QGIS/pull/4113

**Comment by PeterPetrik** [comment]
<img width="1111" alt="Screenshot 2020-08-20 at 00 17 48" src="https://user-images.githubusercontent.com/804608/90758382-3a2f8a80-e2df-11ea-99c4-dc0b54379ffe.png">

**Comment by PeterPetrik** [comment]
If forced to take libraries from bundled MacOS/lib, it fails to load some Apple's native frameworks due to symbol/lib clashes (libjpeg, libpng, libtiff and libsqlite3)
```
peter@pp-mb:/opt/QGIS/qgis-dev$ DYLD_LIBRARY_PATH=/opt/QGIS/qgis-dev/bundle/QGIS.app/Contents/MacOS/lib ./Designer.app/Contents/MacOS/Designer 
dyld: Symbol not found: __cg_jpeg_resync_to_restart
  Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
  Expected in: /opt/QGIS/qgis-dev/bundle/QGIS.app/Contents/MacOS/lib/libJPEG.dylib
 in /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
Abort trap: 6
```

if I set rpath to bundled lib (equivallent of setting fallback library path), the /usr/lib/libsqlite3.dylib is picked which causes coredump when using `QgsProject::instance()` (see previous screen)
```
peter@pp-mb:/opt/QGIS/qgis-dev$ DYLD_FALLBACK_LIBRARY_PATH=/opt/QGIS/qgis-dev/bundle/QGIS.app/Contents/MacOS/lib ./Designer.app/Contents/MacOS/Designer 
src/core/auth/qgsauthmanager.cpp:3711 : (authDbQuery) [1ms] Auth db query exec() FAILED
WARNING: Auth db query exec() FAILED
Segmentation fault: 11
```

**Comment by PeterPetrik** [comment]
fixed by usage of DYLD_INSERT_LIBRARIES

the Designer must be lauched by wrapper script located in `QGIS.app/Contents/MacOS/bin/designer`
